### DISCUSSION TITLE
开源项目的困境

### DISCUSSION THREAD

**Discussion by Innei** [Innei/Shiro#318] [General]
最近，开源圈出现了不少的话题。有关于开源和商业化的伪命题，有开源项目被剽窃用于商业化，有对开源项目的深刻思考等。

我想很多不是以公司名义发布的个人开发者开源项目都是非盈利的项目，大多都是开发者自身需要才编写的，然后顺便开源出来回馈依托的开源社区。我也是一样。

但是，开源并不盈利，却要花费大量的时间去支持。甚至不妨被小人剽窃贴上自身的标签，或者被用于商业化盈利，或者还有人很多的伸手党等等。

我害怕这样，也不想花费的精力都是徒劳。因此下一步，我将计划闭源开发此项目，当然此开源项目不会删除，但是不会再迭代任何新功能。

**Comment by BLxcwg666** [comment]
怎么说呢，其实并不是每个人都是你想的那样，`被用于商业化盈利`确实是一方面的问题  
我不是很会安慰别人
但是还请别忘了一直在默默支持你的人  
有时候没必要把这些方面看的太重，做自己喜欢的事就好  

~~咱就是说要不去整个 国家认证软件著作权证书 什么的~~

**Reply by BLxcwg666** [reply]
~~其实就是自己想用，上面的话可以当放屁了~~

**Reply by ttimochan** [reply]
> ~~咱就是说要不去整个 国家认证软件著作权证书 什么的~~


软著的意义就是方便打官司，然而个人没有精力没有时间去弄这个。

**Comment by WetABQ** [comment]
<img width="2338" alt="image" src="https://github.com/Innei/Shiro/assets/25319400/97e5775a-4fae-4418-b5cd-857f2652930a">

**Reply by ttimochan** [reply]
这个仓库不会删，但是后续更新闭源

**Reply by Innei** [reply]
可以用来盈利 你可以在分发软件的时候收费，但你必须在收费前向你的客户提供该软件的 GNU GPL 许可协议，以便让他们知道，他们可以从别的渠道免费得到这份软件，以及你收费的理由。
可自由修改 如果你想添加或删除某个功能，没问题，如果你想在别的项目中使用部分代码，也没问题，唯一的要求是，使用了这段代码的项目也必须使用 GPL 协议。

**Reply by WetABQ** [reply]
我不知道，但是我认为本帖讨论的商业化完全在 GPL 的许可范围下，并且也没规定在应用启动时要强制展示或者怎么样的，我把协议原文放在一个不为人所知的角落也不算违反协议。

**Reply by Innei** [reply]
v2ex.com/t/1024662

**Reply by Innei** [reply]
咬文嚼字，这句话和此项目无关。

**Reply by WetABQ** [reply]
这是你在开始这个项目时自己的选择，你选择了将其发布在 Github 上，选择了 AGPL 协议，那就要接受被商业化的事实。当然，商业化的人选择不选择公布你的协议原文就是另一个故事了，单论商业化，这是被协议允许的内容，并不是什么咬文嚼字，这是你的选择，你可以闭源。

事实上：即便我现在重新编译 VSCode 并且打包成安装包收费，只要不使用其 Logo。这种做法并不违反任何协议。

**Reply by Innei** [reply]
如果你说这个协议商用，并且修改了源代码但是没有开放源码就因为违反了。一般商用的套皮谁会开源

**Reply by Innei** [reply]
当然我不是说什么为不违反这件事，我只是觉得恶心

**Reply by WetABQ** [reply]
你选择了开源，选择了 AGPL，你在道德上没有攻击商业化行为的意义。当然你可以选择 Copyright all reserved，也可以选择闭源。

虽然在你提到的这件事上我只是在说 AGPL 允许商业化，但是我的个人立场确实是反对任何 GNU 协议。
本质上是对使用者 / 贡献者的强奸，是一种对开源自由的控制，这和 CCP 没有任何本质的区别，强调一种细致入微的控制。这种宣称的「自由」软件，反而剥夺了作为开发者选择许可证的自由，何尝不是一种暴政和倒退。

当然这是只是个人的观点，权当笑料。你当然可以抨击资本，抨击小人，我没意见，我只是路过，想告诉你，AGPL，其实允许商用。
